src/daemon.c

```diff
@@ -181,8 +181,10 @@ int main(int argc, char *argv[])
 	if ((pid = fork()) < SUCCESS) {
 		rc = pid;
 		goto exit_on_failure;
-	} else if (pid != SUCCESS)
+	} else if (pid != SUCCESS) {
+		if (path) free(path);
```

Comment:
I believe `args` should be freed too.

Reply:
I made a mistake here. We shouldn't be freeing anything because we are terminating the parent. I was too quick to push this PR. I reverted bak to the original code because here we are officially turning the process into a daemon by detaching from the parent PID.